src/main/java/seedu/address/logic/commands/CommandUtil.java

```diff
@@ -0,0 +1,41 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.commands.Command.MESSAGE_ADDRESS_BOOK_COMMAND_PREREQUISITE;
+
+import java.util.List;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+
+/**
+ * Contains utility methods for the Command classes.
+ */
+public class CommandUtil {
+    /**
+     * Prepares the model for index based commands.
+     *
+     * @param model Model to be used.
+     * @param index Index of the patient.
+     * @throws CommandException If command input is invalid.
+     */
+    public static List<Person> prepareFilteredList(Model model, Index index) throws CommandException {
+        requireNonNull(model);
+
+        if (model.isRecordListDisplayed()) {
+            throw new CommandException(MESSAGE_ADDRESS_BOOK_COMMAND_PREREQUISITE);
+        }
+
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        // Check if index given is out of bounds
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        return lastShownList;
+    }
```

Comment:
The content of this method `CommandUtil#prepareFilteredList()` is indeed repeated in the `execute()` method of multiple Command classes, but should it be extracted out? Since this method performs many different actions, such as:
- Checking if record list is displayed,
- Getting the filtered person list,
- Checking if the given `Index` is out of bounds,
- Returning the filtered person list

Perhaps the name of this method does not truly reflect the different actions that it performs?

Reply:
I feel that it's fine to extract it since it was exactly the same for all the different commands, it was pretty much just checks, but maybe the return part of the method can be removed if you think it's really unsuitable to be there.